stellar-contract-env-common/src/symbol.rs

```diff
@@ -282,6 +282,46 @@ impl FromIterator<char> for Symbol {
     }
 }
 
+use crate::xdr::ScVal;
+
+#[cfg(feature = "std")]
+impl TryFrom<ScVal> for Symbol {
+    type Error = ();
+    fn try_from(v: ScVal) -> Result<Self, Self::Error> {
+        (&v).try_into()
+    }
+}
+#[cfg(feature = "std")]
+impl TryFrom<&ScVal> for Symbol {
+    type Error = ();
+    fn try_from(v: &ScVal) -> Result<Self, Self::Error> {
+        if let ScVal::Symbol(vec) = v {
+            vec.try_into().map_err(|_| ())
+        } else {
+            Err(())
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+impl TryFrom<Symbol> for ScVal {
+    type Error = ();
+    fn try_from(s: Symbol) -> Result<Self, Self::Error> {
+        let res: Result<Vec<u8>, _> = s.into_iter().map(<u8 as TryFrom<char>>::try_from).collect();
+        Ok(ScVal::Symbol(
+            res.map_err(|_| ())?.try_into().map_err(|_| ())?,
+        ))
+    }
+}
+
+#[cfg(feature = "std")]
+impl TryFrom<&Symbol> for ScVal {
+    type Error = ();
+    fn try_from(s: &Symbol) -> Result<Self, Self::Error> {
+        s.clone().try_into()
+    }
```

Comment:
I usually try to do these in reverse, implement for the ref version, and then have the other one cast to it. You're collecting in the non-ref version anyway, so then the ref version becomes 2x clones.

Reply:
Does the reverse work in this case? I'm calling `into_iter` which takes ownership. Is there a non-into option here?